### ISSUE TITLE
My model outputs the same value for all data during prediction

### ISSUE THREAD

**Issue by YuvalSilman** [keras-team/keras#13891]
Hi,
I am using a simple shallow model for a regression task.
When using a single neuron (that outputs the continuous result) i get reasonable results both while training and while predicting.
When a add another layer with few neurons before the last layer, althought the model's training seems normal, when i am using the model to predict i get the same constant value for all inputs.
In addition, i tried to change the activation function of the added layer to linear activation and as expected i get reasonable results (the same like in the case of only 1 neuron as output).

Has anyone encountered such problem and knows how to fix it?

**Comment by eduardofv** [comment]
Have you tried varying the learning rate, maybe making it smaller?

**Comment by YuvalSilman** [comment]
> Have you tried varying the learning rate, maybe making it smaller?

Yes, i have used different scales of learning rate, as well i was using reduce learning rate on plateu. Since i wrote the post i have tried to use tanh activation and found out that it works fine with it (i still prefere to use relu). So i am really lost because from my prespective it looks like the problem is in the activation function, which doesnt seem reasonable.

**Comment by fernandonieuwveldt** [comment]
Can you post some code snippets?

**Comment by eduardofv** [comment]
@YuvalSilman Maybe running a [learning rate finder](https://www.pyimagesearch.com/2019/08/05/keras-learning-rate-finder/) in your model and data can give you a better sense of what can you expect. If it's totally flat it may indicate a problem with your data. In similar situation I've found replacing Adam with RMSprop helps. 

Agree that a snippet will help, maybe it's some subtle bug with the data/model handling.

**Comment by YuvalSilman** [comment]
![code snip](https://user-images.githubusercontent.com/49240261/76698797-b62e0280-66af-11ea-8874-489edd899dc2.png)

I attached the code which builds the model and my training function.
@eduardofv : I will try using RMSprop, thanks. The prediction is totally flat, the training is fine and looks normal.

**Comment by eduardofv** [comment]
Well, I can't find something in your code... are you normalizing your inputs? and if you do, does the data used on predictions are normalized the same way?

**Comment by YuvalSilman** [comment]
Hi, the data is not normalized. Do you think it might be the source of the problem?

**Comment by Saduf2019** [comment]
@YuvalSilman 
Moving this issue to closed status as there has been no activity, in case you still face the error please create a new issue.
